test/e2e/framework/framework.go

```diff
@@ -392,7 +395,14 @@ func newSingleIngress(name, path, host, ns, service string, port int, annotation
 					},
 				},
 			},
-		},
+		}
+	} else {
+		ing.Spec = extensions.IngressSpec{
+			Backend: &extensions.IngressBackend{
```

Comment:
It might be clearer to have a catch-all backend style ingress be its own function rather than toggle based on the host